lib/real-device-management.js

```diff
@@ -72,15 +73,71 @@ async function installToRealDevice(device, app, bundleId, opts) {
     return;
   }
 
-  const {skipUninstall, strategy, timeout} = opts ?? {};
+  const {skipUninstall, strategy, timeout, shouldEnforceUninstall} = opts ?? {};
+  const shouldUninstallApp = !skipUninstall && bundleId;
 
-  if (!skipUninstall && bundleId && (await device.isAppInstalled(bundleId))) {
+  if (shouldUninstallApp) {
     log.debug(`Reset requested. Removing app with id '${bundleId}' from the device`);
     await device.remove(bundleId);
   }
   log.debug(`Installing '${app}' on device with UUID '${device.udid}'...`);
-  await device.install(app, timeout, strategy);
-  log.debug('The app has been installed successfully.');
+
+  try {
+    await device.install(app, timeout, strategy);
+    log.debug('The app has been installed successfully.');
+  } catch (e) {
+    // Do nothing if the session did not require to enforce uninstalling the application
+    if (!shouldEnforceUninstall) {
+      throw e;
+    }
+
+    if (!bundleId) {
```

Comment:
I assume this won't happen. If the app path was given then we can always read the bundle identifier

Reply:
Yea, but the lint claimed the input could be undefined. Let me check typehint

Reply:
I have updated `unfedined` usage in the caller side in https://github.com/appium/appium-xcuitest-driver/pull/2050/commits/00ed99fdd75d54d74ab1971f5998ed2173b8b94d

Reply:
The changes is now in https://github.com/appium/appium-xcuitest-driver/pull/2054